lib/puppet/type/cisco_bgp_af.rb

```diff
@@ -179,21 +191,153 @@ def self.title_patterns
   # Properties #
   ##############
 
-  validate do
-    fail("The 'asn' parameter must be set in the manifest.") if self[:asn].nil?
-    fail("The 'afi' parameter must be set in the manifest.") if self[:afi].nil?
-    fail("The 'safi' parameter must be set in the manifest.") if self[:safi].nil?
-  end
+  newproperty(:additional_paths_install) do
+    desc 'install a backup path into the forwarding table and provide prefix ' \
+         'independent convergence (PIC) in case of a PE-CE link failure. ' \
+         "Valid values are true, false, or 'default'"
+
+    newvalues(:true, :false, :default)
+  end # property additional_paths_install
+
+  newproperty(:additional_paths_receive) do
+    desc 'Enables the receive capability of additional paths for all' \
+      'of the neighbors under this address family for which the ' \
+      'capability has not been disabled. ' \
+      "Valid values are true, false, or 'default'"
+
+    newvalues(:true, :false, :default)
+  end # property additional_paths_receive
+
+  newproperty(:additional_paths_selection) do
+    desc 'Configures the capability of selecting additional paths for a prefix. ' \
+      'Valid values are a string defining the name of the route-map.'
+
+    validate do |value|
+      fail("'additional_paths_selection' value must be a string") unless
+        value.is_a? String
+    end
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property additional_paths_selection
+
+  newproperty(:additional_paths_send) do
+    desc 'Enables the send capability of additional paths for all of the ' \
+      'neighbors under this address family for which the capability has ' \
+      "not been disabled. Valid values are true, false, or 'default'"
+
+    newvalues(:true, :false, :default)
+  end # property additional_paths_send
 
   newproperty(:client_to_client) do
-    desc "client_to_client. Valid values are true, false, or 'default'"
+    desc 'Configure client-to-client route reflection. Valid values are ' \
+         "true, false, or 'default'"
 
     newvalues(:true, :false, :default)
   end # property client_to_client
 
+  newproperty(:dampen_igp_metric) do
+    desc "Specify dampen value for IGP metric-related changes, in seconds.
+          Valid values are Integer, keyword 'default'."
```

Comment:
I think you'll want to format these like so:

```
    desc 'Specify dampen value for IGP metric-related changes, in seconds. ' \
          "Valid values are Integer, keyword 'default'."
```

...otherwise I think they'll display with a big space in front of each sentence.